app/Database/Repositories/UserRepositoryInterface.php

```diff
@@ -24,7 +24,21 @@
      * @return mixed
      */
     public function all($sortBy = 'created_at', $sortDir = 'desc', $perPage = 10);
-
+    
+    /**
+     * Get all users who match a certain set of criteria.
+     *
+     * @param string username
+     * @param string email
+     * @param int role_id
+     * @param string $sortBy
+     * @param string $sortDir
+     * @param int $perPage
+     *
+     * @return mixed
+     */
+    public function search($username = "", $email = "", int $role_id = 0, $sortBy = 'created_at', $sortDir = 'asc', $perPage = 10);
```

Comment:
Again, same as before: `role_id` should be `roleId` and can you please type hint the other parameters?